software/main_w_ref.py

```diff
@@ -0,0 +1,196 @@
+    cam = camera.RealsenseCamera(exposure = 100)
+    processor = image_processor.ImageProcessor(cam, debug=debug)
+    processor.start()
+    start = time.time()
+    fps = 0
+    frame = 0
+    frame_cnt = 0
+    basket = processedData.basket_b
+    
+     # praegu hard code 
+     
+async def refcommClient(robotState, attackingSide):
+    robotName = 'Valdis'
+    connectTo = 'ws://172.17.155.146:8222/'
+    async with websockets.connect(connectTo, ping_interval=None) as websocket:
+        refcomm = await websocket.recv()
+        print('Signal recieved...')
+        refdict = json.loads(refcomm)
+        print(refdict['targets'])
+        if refdict['signal'] == 'start':
+            print(refdict['baskets'])
+            signal = True
+        if robotName in refdict['targets']:
+            if refdict['signal'] == 'start':
+                if refdict['baskets'][refdict['targets'].index(robotName)] == 'blue':
+                    basket = processedData.basket_b
+                elif refdict['baskets'][refdict['targets'].index(robotName)] == 'magenta':
+                    basket = processedData.basket_m
+                STATE = Machine_state.FIND_BALL
+                print('Starting competition!')
+
+            elif refdict['signal'] == 'stop':
+                i.send_to_robot(0,0,0,0)
+                i.close
+                print('Stopping competition!')
+        else:
+            print('Signal not directed at', robotName)
+        
+        
+
+            #print(robotState.value, attackingSide)
+
+
+def refclient(STATE, basket):
+    return asyncio.run(refcommClient(STATE, basket))
+
+if __name__ == '__main__':
+    debug = False
+    cam = camera.RealsenseCamera(exposure = 100)
+    processor = image_processor.ImageProcessor(cam, debug=debug)
+    processor.start()
+    
+    
+    try:
+        while True:
+            break
+            # if signal == start (continue)
+            # else break
+            # has argument aligned_depth that enables depth frame to color frame alignment. Costs performance
+            if STATE is Machine_state.THROW_BALL:
+                processedData = processor.process_frame(aligned_depth=True)
+            else:
+                processedData = processor.process_frame(aligned_depth=False)
+           
+            # This is where you add the driving behaviour of your robot. It should be able to filter out
+            # objects of interest and calculate the required motion for reaching the objects
+
+            frame += 1
+            if frame % 15 == 0:
+                frame = 0
+                end = time.time()
+                fps = 30 / (end - start)
+                start = end
+                #print("FPS: {}, framecount: {}".format(fps, frame_cnt))
+                if len(processedData.balls) > 0:
+                    # on pall
+                    ball = processedData.balls[-1]
+                    if STATE is Machine_state.FIND_BALL or STATE is Machine_state.MOVE_TO_BALL:
```

Comment:
Nothing should be in the `if frame % 15 == 0` loop. It is just to slow down the FPS spam into console.